server/src/main/java/io/crate/session/Session.java

```diff
@@ -614,6 +616,21 @@ public CompletableFuture<?> sync() {
         }
     }
 
+    public List<Statement> simpleQuery(String queryString) {
+        try {
+            return SqlParser.createStatementsForSimpleQuery(
+                queryString,
+                str -> PgArrayParser.parse(
+                    str,
+                    bytes -> new String(bytes, StandardCharsets.UTF_8)
+                )
+            );
+        } catch (Throwable t) {
+            jobsLogs.logPreExecutionFailure(UUIDs.dirtyUUID(), queryString, SQLExceptions.messageOf(t), sessionSettings.sessionUser());
+            throw t;
+        }
```

Comment:
I was wondering if we could `reuse session.parse` here
with  `session.parse(UNNAMED, stmt, emptyList())` but not sure it worth it, we would have to take care of `bind`?

Reply:
I don't follow... `parse` does not use `SqlParser.createStatementsForSimpleQuery` could you clarify?

Reply:
I was thinking about using  `statements = SqlParser.createStatements(queryString)` and then call `session.parse` for each statement but looks like we intentionally stopped using `SqlParser.createStatements(queryString)` due to a bug, see https://github.com/crate/crate/commit/397ac37ce3d91a41f0bfd366d141f1ebfe1d96df